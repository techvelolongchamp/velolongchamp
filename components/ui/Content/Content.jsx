import React from 'react'
import PropTypes from 'prop-types'

import { Wrapper } from './Content.styled'

const Content = ({ children }) => <Wrapper>{children}</Wrapper>

Content.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

export default Content
