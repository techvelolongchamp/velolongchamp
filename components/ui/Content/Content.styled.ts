import styled from 'styled-components'

export const Wrapper = styled.div`
  margin: 0 auto;
  max-width: ${({ theme }) => theme.contentMaxWidth};
  padding: ${({ theme }) => theme.spacing.s};
  height: inherit;
`
