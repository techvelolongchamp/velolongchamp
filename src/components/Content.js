import React from 'react'
import styled from 'styled-components'

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: ${({ theme }) => theme.contentMaxWidth};
  padding: ${({ theme }) => theme.spacing.s};
  height: inherit;
`

const Content = ({ children }) => <Wrapper>{children}</Wrapper>

export default Content
