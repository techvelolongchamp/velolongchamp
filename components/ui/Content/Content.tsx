import React from 'react'

import { Wrapper } from './Content.styled'

const Content: React.FC<{ maxWidth?: string; children: React.ReactNode }> = ({
  maxWidth,
  children,
}) => <Wrapper maxWidth={maxWidth}>{children}</Wrapper>

export default Content
