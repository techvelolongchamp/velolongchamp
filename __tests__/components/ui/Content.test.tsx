import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import Content from '../../../components/ui/Content/Content'

describe('Content unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<Content>Hello world</Content>)

    expect(container).toMatchSnapshot()
  })
})
