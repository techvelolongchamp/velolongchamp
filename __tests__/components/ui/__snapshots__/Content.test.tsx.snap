// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content unit tests should render correctly 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

<div>
  <div
    class="c0"
  >
    Hello world
  </div>
</div>
`;
